### ISSUE TITLE
request.files[...].save() is not compatible with Trio

### ISSUE THREAD

**Issue by lordmauve** [pgjones/quart-trio#7]
Quart [uses aiofiles to write uploaded files to the filesystem](https://github.com/pallets/quart/blob/3cba3091daf5dfe53135ab4b45d533a4793523be/src/quart/datastructures.py#L32), e.g. using code like

```python
from quart import request

...

    files = await request.files
    files['content'].save(filename)
```

However aiofiles does not support Trio, so this raises RuntimeError when running with quart-trio:

```
  File "/home/mauve/x/.venv/lib/python3.11/site-packages/quart/datastructures.py", line 32, in save
    async with async_open(destination, "wb") as file_:
  File "/home/mauve/x/.venv/lib/python3.11/site-packages/aiofiles/base.py", line 98, in __aenter__
    self._obj = await self._coro
                ^^^^^^^^^^^^^^^^
  File "/home/mauve/x/.venv/lib/python3.11/site-packages/aiofiles/threadpool/__init__.py", line 85, in _open
    loop = asyncio.get_running_loop()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: no running event loop
```

**Comment by CoolCat467** [comment]
Needs to change this line:
https://github.com/pallets/quart/blob/3cba3091daf5dfe53135ab4b45d533a4793523be/src/quart/formparser.py#L42

I think I see `FileStorage` class being used for type annotations and in testing, but not sure how you would go about patching that.

**Comment by pgjones** [comment]
Fixed by f9a0281e4d3d953a84dce9933f687f8dcd7a3cfc
